File: Atividade_contextualizacao/ClienteJava/src/tsi/webservices/comunicacaoSocket/modelos/Turma.java
package tsi.webservices.comunicacaoSocket.modelos;

import java.util.ArrayList;
import java.util.List;

public class Turma {

	private static int id = 0;

	private int idTurma;
	private int ano;
	private String curso;
	private List<Aluno> alunos;

	public Turma(int ano, String curso) {
		id++;
		this.idTurma = id;
		this.ano = ano;
		this.curso = curso;
		this.alunos = new ArrayList<Aluno>();
	}

	public int getId() {
		return idTurma;
	}

	public void setId(int idTurma) {
		this.idTurma = idTurma;
	}

	public int getAno() {
		return ano;
	}

	public void setAno(int ano) {
		this.ano = ano;
	}

	public String getCurso() {
		return curso;
	}

	public void setCurso(String curso) {
		this.curso = curso;
	}

	public List<Aluno> getAlunos() {
		return alunos;
	}

	public void setAlunos(List<Aluno> alunos) {
		this.alunos = alunos;
	}

	public void adicionarAluno(Aluno aluno) {
		alunos.add(aluno);
	}

	@Override
	public String toString() {
		return "Turma: id= " + idTurma + " ano= " + ano + ", curso= " + curso + ", alunos= " + alunos;
	}

}

File: Atividade_contextualizacao/ServerPython/Turma.py
from Aluno import Aluno

class Turma:

    def __init__(self,id,curso,ano):
        self.id = id
        self.ano = ano
        self.curso = curso
        self.alunos = []
        
    
    def getId(self):
        return self.id
    
    def setId(self,id):
        self.id = id
    
    def getAno(self):
        return self.ano

    def setAno(self,ano):
        self.ano = ano

    def getCurso(self):
        return self.curso

    def setCurso(self,curso):
        self.curso = curso

    def  setAluno(self,aluno):
        self.alunos.append(aluno)
        
    def printaAlunos(self):
        message = ""

        for aluno in self.alunos:
            message = message + aluno.__srt__() + " "
            

        return message

    def __srt__(self):
        return "{} {} {} {}".format(self.id, self.ano, self.curso, self.printaAlunos())
    
    def alunosMatriculados(self):
        contador = 0

        for aluno in self.alunos:
            if (aluno.getMatriculado() == "1"):
                contador+=1

        return contador
    
    def resposta(self):
        return "A turma de código {} de {} do curso {} possui {} alunos, dos quais {} estão devidamente matriculados".format(self.id, self.ano ,self.curso, len(self.alunos), self.alunosMatriculados())